r/tests/testthat/test-record-batch-reader.R

```diff
@@ -157,3 +157,31 @@ test_that("reader head method edge cases", {
   expect_equal(head(reader, 0), Table$create(x = integer(0), y = character(0)))
   expect_equal(head(reader, 100), Table$create(batch, batch))
 })
+
+test_that("RBR methods", {
+  batch <- record_batch(
+    x = 1:10,
+    y = letters[1:10]
+  )
+  sink <- BufferOutputStream$create()
+  writer <- RecordBatchStreamWriter$create(sink, batch$schema)
+  writer$write(batch)
+  writer$write(batch)
+  writer$close()
+  buf <- sink$finish()
+
+  reader <- RecordBatchStreamReader$create(buf)
+  expect_output(
+    print(reader),
+    "RecordBatchStreamReader
+x: int32
+y: string"
+  )
```

Comment:
I will say, this kind of test is a good usecase for `expect_snapshot()`. We don't need to change it, but it makes looking at the output much nicer than dealing with embedded newlines + tabs.